explorer/interpreter/type_checker.cpp

```diff
@@ -2614,7 +2657,8 @@ auto TypeChecker::LookupRewriteInWitness(
     CARBON_ASSIGN_OR_RETURN(
         Nonnull<const Value*> constraint,
         Substitute(impl_witness->bindings(),
-                   impl_witness->declaration().constraint_type()));
+                   impl_witness->declaration().constraint_type(),
+                   impl_witness->declaration().source_loc()));
```

Comment:
I think we should pass into this function the location where the lookup was triggered (the location of the member access expression).